runtime/runtime/src/congestion_control.rs

```diff
@@ -26,8 +26,19 @@ use near_store::{StorageError, TrieAccess, TrieUpdate};
 use std::borrow::Cow;
 use std::collections::HashMap;
 
-pub enum ReceiptSink {
-    V2(ReceiptSinkV2),
+pub(crate) enum ReceiptSink {
+    V2(ReceiptSinkV2WithInfo),
+}
+
+pub(crate) struct ReceiptSinkV2WithInfo {
```

Comment:
optional nit: maybe can add a comment about this type existing because it helps avoid getting the shard layout many times.

Reply:
I think describing "what the code did before to motivate introduction of the current code" is largely an uninteresting minutiae that detracts from the current code structure. If anybody cares they can find the motivation in `git blame` and the commit message (which always should describe why a change is being made, as is done here.)

But I did add a comment to explain why this type exists in the current iteration of the code base (over, say, flattening it into a single struct.)